Rubberduck.Core/UI/CodeExplorer/Commands/ImportCommand.cs

```diff
@@ -86,9 +93,66 @@ protected override void OnExecute(object parameter)
 
             foreach (var filename in _openFileDialog.FileNames)
             {
-                using (var components = project.VBComponents)
+                var extension = Path.GetExtension(filename);
+
+                var sourceText = string.Join(Environment.NewLine, File.ReadAllLines(filename));
+                var tempHelper = (CodeExplorerItemViewModel)parameter;
+                var newFolderName = (parameter is CodeExplorerCustomFolderViewModel) ? tempHelper.Name : tempHelper.GetSelectedDeclaration().CustomFolder;
+
+                var startRule = VBACodeStringParser.Parse(sourceText, t => t.startRule());
+                
+                if (HasModuleDeclarations((Antlr4.Runtime.ParserRuleContext)startRule.parseTree, out var moduleDeclarations))
+                {
+                    if (HasOptionExplicit((Antlr4.Runtime.ParserRuleContext)startRule.parseTree, out var optionExplicitStmt))
+                    {
+                        if (HasFolderAnnotation((Antlr4.Runtime.ParserRuleContext)startRule.parseTree, out var FolderAnnotation))
```

Comment:
Why not first ask for the folder annotation and then for Option Explicit? The latter is only used in case there is no folder.

Reply:
Included @Hosch250's suggested edit and it resolved this.